refactor/src/Cage.py

```diff
@@ -242,17 +178,20 @@ def update(self, cur_date: dt.datetime, step_size: int, pressure: int) -> tuple:
             num_infection_events,
             lice_from_reservoir,
             avail_dams_batch,
-            new_egg_batch,
             new_offspring_distrib,
             returned_dams,
             hatched_arrivals_dist
         )
 
-        self.logger.debug("\t\tfinal fish population = {}".format(self.num_fish))
         self.logger.debug("\t\tfinal lice population= {}".format(self.lice_population))
-        egg_distrib = new_egg_batch.geno_distrib
-        hatch_date = new_egg_batch.hatch_date
-        self.logger.debug("\t\tlice offspring = {}".format(sum(egg_distrib.values())))
+
+        egg_distrib = {}    # type: GenoDistrib
```

Comment:
Note: this is actually a review for line 100, but there is something to say about this line as well.

Firstly: `tuple` is a type constructor, not a type annotation. I think the wished return type for `Cage.update()` is `Tuple[GenoDistrib, Optional[dt.datetime]]`. In lack of a proper type annotation pytype may give warnings or even errors. I believe if the proper return type is given there is no need to put a `# type: GenoDistrib` here. Similarly for the line below. I am not sure how this slipped from my last PR. My bad.

In general, I recommend running stuff through mypy as well. It is way noisier (hence why I did not include it in the CI) but it helps enforcing more compliance.